packages/pigeon/lib/cpp_generator.dart

```diff
@@ -934,22 +1055,33 @@ class CppSourceGenerator extends StructuredGenerator<CppOptions> {
           'flutter::ByteStreamWriter* stream',
         ],
         isConst: true, body: () {
-      if (customTypes.isNotEmpty) {
+      if (enumeratedTypes.isNotEmpty) {
         indent.write(
             'if (const CustomEncodableValue* custom_value = std::get_if<CustomEncodableValue>(&value)) ');
         indent.addScoped('{', '}', () {
-          for (final EnumeratedType customType in customTypes) {
+          for (final EnumeratedType customType in enumeratedTypes) {
+            final String encodeString = customType.type ==
+                    CustomTypes.customClass
+                ? 'std::any_cast<${customType.name}>(*custom_value).ToEncodableList()'
+                : 'static_cast<int>(std::any_cast<${customType.name}>(*custom_value))';
+            final String valueString = customType.enumeration <
+                    maximumCodecFieldKey
+                ? encodeString
+                : 'std::any_cast<$_overflowClassName>(wrap).ToEncodableList()';
```

Comment:
Per offline discussion, this shouldn't be necessary, and I'm surprised it compiles since `wrap` isn't a `std::any`. This string *should* just be `'wrap.ToEncodableList()'` unless I've missed something.